exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/__init__.py

```diff
@@ -60,384 +64,101 @@
 
 The exporter supports the following environment variables for configuration:
 
-:envvar:`OTEL_EXPORTER_ZIPKIN_ENDPOINT`: target to which the exporter will
-send data. This may include a path (e.g. http://example.com:9411/api/v2/spans).
+:envvar:`OTEL_EXPORTER_ZIPKIN_SERVICE_NAME`: Label of the local node in the
+service graph, such as "favstar". Avoid names with variables or unique
+identifiers embedded. Defaults to "unknown". This is a primary label for trace
+lookup and aggregation, so it should be intuitive and consistent. Many use a
+name from service discovery.
+
+:envvar:`OTEL_EXPORTER_ZIPKIN_ENDPOINT`: zipkin collector endpoint to which the
+exporter will send data. This may include a path (e.g.
+http://example.com:9411/api/v2/spans).
 
-:envvar:`OTEL_EXPORTER_ZIPKIN_TRANSPORT_FORMAT`: transport interchange format
-to use when sending data. Currently only Zipkin's v2 json and protobuf formats
-are supported, with v2 json being the default.
+:envvar:`OTEL_EXPORTER_ZIPKIN_ENCODING`: transport interchange format
+encoder to use when sending data. Refer to
+opentelemetry.exporter.zipkin.encoder.Encoding for supported options.
 
 API
 ---
 """
 
-import json
 import logging
-from typing import Optional, Sequence, Union
-from urllib.parse import urlparse
+from typing import Optional, Sequence
 
 import requests
 
 from opentelemetry.configuration import Configuration
-from opentelemetry.exporter.zipkin.gen import zipkin_pb2
+from opentelemetry.exporter.zipkin.encoder import Encoder, Encoding
+from opentelemetry.exporter.zipkin.encoder.v1.json import JsonV1Encoder
+from opentelemetry.exporter.zipkin.encoder.v2.json import JsonV2Encoder
+from opentelemetry.exporter.zipkin.encoder.v2.protobuf import ProtobufEncoder
+from opentelemetry.exporter.zipkin.node_endpoint import IpInput, NodeEndpoint
 from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
-from opentelemetry.trace import Span, SpanContext, SpanKind
-from opentelemetry.trace.status import StatusCode
-
-TRANSPORT_FORMAT_JSON = "json"
-TRANSPORT_FORMAT_PROTOBUF = "protobuf"
-
-DEFAULT_RETRY = False
-DEFAULT_URL = "http://localhost:9411/api/v2/spans"
-DEFAULT_MAX_TAG_VALUE_LENGTH = 128
-
-SPAN_KIND_MAP_JSON = {
-    SpanKind.INTERNAL: None,
-    SpanKind.SERVER: "SERVER",
-    SpanKind.CLIENT: "CLIENT",
-    SpanKind.PRODUCER: "PRODUCER",
-    SpanKind.CONSUMER: "CONSUMER",
-}
+from opentelemetry.trace import Span
 
-SPAN_KIND_MAP_PROTOBUF = {
-    SpanKind.INTERNAL: zipkin_pb2.Span.Kind.SPAN_KIND_UNSPECIFIED,
-    SpanKind.SERVER: zipkin_pb2.Span.Kind.SERVER,
-    SpanKind.CLIENT: zipkin_pb2.Span.Kind.CLIENT,
-    SpanKind.PRODUCER: zipkin_pb2.Span.Kind.PRODUCER,
-    SpanKind.CONSUMER: zipkin_pb2.Span.Kind.CONSUMER,
-}
-
-NAME_KEY = "otel.library.name"
-VERSION_KEY = "otel.library.version"
-
-SUCCESS_STATUS_CODES = (200, 202)
+DEFAULT_SERVICE_NAME = "unknown"
+DEFAULT_ENDPOINT = "http://localhost:9411/api/v2/spans"
+DEFAULT_ENCODING = Encoding.V2_JSON
```

Comment:
Should this be the default? I couldn't find any default mentioned in the spec clearly but the closest I could find is PROTOBUF from [Exporter Selection](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#exporter-selection)

Reply:
I opted for this to maintain backwards compatibility since V2 JSON was the first encoding supported.

Reply:
circling back - I've removed the default and made encoding (now protocol) a mandatory arg to align with spec